packages/visx-xychart/src/components/series/private/BaseLineSeries.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext, useCallback } from 'react';
+import LinePath from '@visx/shape/lib/shapes/LinePath';
+import { AxisScale } from '@visx/axis';
+import DataContext from '../../../context/DataContext';
+import { SeriesProps } from '../../../types';
+import withRegisteredData, { WithRegisteredDataProps } from '../../../enhancers/withRegisteredData';
+import getScaledValueFactory from '../../../utils/getScaledValueFactory';
+import useEventEmitter, { HandlerParams } from '../../../hooks/useEventEmitter';
+import findNearestDatumX from '../../../utils/findNearestDatumX';
+import TooltipContext from '../../../context/TooltipContext';
+import findNearestDatumY from '../../../utils/findNearestDatumY';
+
+export type BaseLineSeriesProps<
+  XScale extends AxisScale,
+  YScale extends AxisScale,
+  Datum extends object
+> = SeriesProps<XScale, YScale, Datum> & {
+  /** Whether line should be rendered horizontally instead of vertically. */
+  horizontal?: boolean;
+  /** Rendered component which is passed path props by BaseLineSeries after processing. */
+  PathComponent?: React.FC<Omit<React.SVGProps<SVGPathElement>, 'ref'>> | 'path';
+};
+
+function BaseLineSeries<XScale extends AxisScale, YScale extends AxisScale, Datum extends object>({
+  data,
+  dataKey,
+  xAccessor,
+  xScale,
+  yAccessor,
+  yScale,
+  horizontal = true,
+  PathComponent = 'path',
+  ...lineProps
+}: BaseLineSeriesProps<XScale, YScale, Datum> & WithRegisteredDataProps<XScale, YScale, Datum>) {
+  const { colorScale, theme, width, height } = useContext(DataContext);
+  const { showTooltip, hideTooltip } = useContext(TooltipContext) ?? {};
+  const getScaledX = useCallback(getScaledValueFactory(xScale, xAccessor), [xScale, xAccessor]);
+  const getScaledY = useCallback(getScaledValueFactory(yScale, yAccessor), [yScale, yAccessor]);
+  const color = colorScale?.(dataKey) ?? theme?.colors?.[0] ?? '#222';
+
+  const handleMouseMove = useCallback(
+    (params?: HandlerParams) => {
+      const { svgPoint } = params || {};
+      if (svgPoint && width && height && showTooltip) {
+        const datum = (horizontal ? findNearestDatumX : findNearestDatumY)({
+          point: svgPoint,
+          data,
+          xScale,
+          yScale,
+          xAccessor,
+          yAccessor,
+          width,
+          height,
+        });
+        if (datum) {
+          showTooltip({
+            key: dataKey,
+            ...datum,
+            svgPoint,
+          });
+        }
+      }
+    },
+    [dataKey, data, xScale, yScale, xAccessor, yAccessor, width, height, showTooltip, horizontal],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  return (
+    <LinePath
+      data={data}
+      x={getScaledX}
+      y={getScaledY}
+      stroke={color}
+      strokeWidth={2}
+      {...lineProps}
+    >
+      {({ path }) => (
+        <PathComponent stroke={color} strokeWidth={2} {...lineProps} d={path(data) || ''} />
```

Comment:
everything is the same in this file except we use the child render function here. 

sorry I tried to `git mv` and it didn't work in this UI view 💔